src/colorizer/ColorizeCanvas.ts

```diff
@@ -647,6 +650,7 @@ export default class ColorizeCanvas {
     this.onFrameChangeCallback(isMissingFile);
     // Force rescale in case frame dimensions changed
     this.updateScaling(this.dataset?.frameResolution || null, this.canvasResolution);
+    this.currentFrame = index;
```

Comment:
do you want to reset pendingFrame here too maybe?

Reply:
`this.pendingFrame` should already be set to `index`! I don't reset it to a different value when a frame is loaded, so the check for whether a frame is loading in is if `this.pendingFrame !== this.currentFrame`.

Reply:
no i am saying it should be reset to -1 here

Reply:
Sure, I can do that.